torch/csrc/jit/script/ir_emitter.cpp

```diff
@@ -1871,11 +1884,48 @@ struct to_ir {
           {rhs},
           {},
           self);
-
+    } else if (lhsValue->type()->kind() == TypeKind::ClassType) {
+      // Call `__iadd__` so updates happen in place on class types
+      // https://docs.python.org/3/reference/datamodel.html#object.__iadd__
+      const char* magic_method_name;
+      const char* in_place_method_name;
+      const char* out_of_place_method_name;
+      std::tie(in_place_method_name, out_of_place_method_name) =
+          getAugMagicMethod(stmt);
+      const auto rhs = emitExpr(stmt.rhs());
+
+      if (lhsValue->type()->expect<ClassType>()->getMethod(
+              in_place_method_name)) {
+        magic_method_name = in_place_method_name;
+      } else {
+        magic_method_name = out_of_place_method_name;
+      }
+
+      // Insert call to magic method
+      auto sugared_magic_method = makeMagic(
+          magic_method_name,
+          std::make_shared<BuiltinFunction>(
+              getAugOp(stmt, lhsValue->type()), at::nullopt));
```

Comment:
`getAugOp` will never be called or used here. There's not really a need to use the magic method sugared values since within this block it will always be a class type. you can just throw if the class doesn't define `in_place_method_name` or `out_of_place_method_name`.